### PR TITLE
Merged first part of revamp_sign_in_methods

### PR DESCRIPTION
**Pull request by wti806**
Hey there! So you want to contribute to a Firebase SDK? 
Before you file this pull request, please read these guidelines:

### Discussion

  * Read the contribution guidelines (CONTRIBUTING.md).
  * If this has been discussed in an issue, make sure to link to the issue here. 
    If not, go file an issue about this **before creating a pull request** to discuss.

### Testing

  * Make sure all existing tests in the repository pass after your change.
  * If you fixed a bug or added a feature, add a new test to cover your code.

### API Changes

  * At this time we cannot accept changes that affect the public API.  If you'd like to help 
    us make Firebase APIs better, please propose your change in an issue so that we 
    can discuss it together.

### CODE REVIEW

**Comment by wti806** [comment]
Added email address in git config to pass the CLA check

**Comment by bojeil-google** [comment]
Looks good, can you also add the changes to the test app?

File: packages/firebase/externs/firebase-auth-externs.js [hunk 1]
```diff
@@ -1043,7 +1043,59 @@ firebase.auth.Auth.prototype.useDeviceLanguage = function() {};
...
+ * @return {!firebase.Promise<!firebase.auth.UserCredential>}
+ */
+firebase.auth.Auth.prototype.createUserAndRetrieveDataWithEmailAndPassword = function(
+  email,
+  password
+) {};
+
+/**
+ * Creates a new user account associated with the specified email address and
+ * password. Will be depercated, replaced by
```
Comment (bojeil-google) [inline packages/firebase/externs/firebase-auth-externs.js]: This method will be deprecated and will be updated to resolve with a `firebase.auth.UserCredential` as is returned in {@link firebase.auth.Auth#createUserAndRetrieveDataWithEmailAndPassword}.

File: packages/firebase/externs/firebase-auth-externs.js [hunk 2]
```diff
@@ -1382,7 +1434,46 @@ firebase.auth.Auth.prototype.signInAndRetrieveDataWithCredential = function(
...
+ *
+ * @param {string} token The custom token to sign in with.
+ * @return {!firebase.Promise<!firebase.auth.UserCredential>}
+ */
+firebase.auth.Auth.prototype.signInAndRetrieveDataWithCustomToken = function(
+  token
+) {};
+
+/**
+ * Asynchronously signs in using a custom token. Will be depercated, replaced by
```
Comment (bojeil-google) [inline packages/firebase/externs/firebase-auth-externs.js]: This method will be deprecated and will be updated to resolve with a `firebase.auth.UserCredential` as is returned in {@link firebase.auth.Auth#signInAndRetrieveDataWithCustomToken}.

File: packages/firebase/externs/firebase-auth-externs.js [hunk 3]
```diff
@@ -1416,7 +1507,59 @@ firebase.auth.Auth.prototype.signInAndRetrieveDataWithCredential = function(
...
+ * @param {string} password The users password.
+ * @return {!firebase.Promise<!firebase.auth.UserCredential>}
+ */
+firebase.auth.Auth.prototype.signInAndRetrieveDataWithEmailAndPassword = function(
+  email,
+  password
+) {};
+
+/**
+ * Asynchronously signs in using an email and password. Will be depercated,
```
Comment (bojeil-google) [inline packages/firebase/externs/firebase-auth-externs.js]: This method will be deprecated and will be updated to resolve with a `firebase.auth.UserCredential` as is returned in {@link firebase.auth.Auth#signInAndRetrieveDataWithEmailAndPassword}.

File: packages/firebase/externs/firebase-auth-externs.js [hunk 4]
```diff
@@ -1551,7 +1694,40 @@ firebase.auth.ConfirmationResult.prototype.confirm = function(
...
+ *     console.error(error);
+ *   }
+ * });
+ *
+ * @return {!firebase.Promise<!firebase.auth.UserCredential>}
+ */
+firebase.auth.Auth.prototype.signInAnonymouslyAndRetrieveData = function() {};
+
+/**
+ * Asynchronously signs in as an anonymous user. Will be depercated, replaced
```
Comment (bojeil-google) [inline packages/firebase/externs/firebase-auth-externs.js]: This method will be deprecated and will be updated to resolve with a `firebase.auth.UserCredential` as is returned in {@link firebase.auth.Auth#signInAnonymouslyAndRetrieveData}.

File: packages/firebase/externs/firebase-auth-externs.js [hunk 5]
```diff
@@ -1043,7 +1043,59 @@ firebase.auth.Auth.prototype.useDeviceLanguage = function() {};
 
 /**
  * Creates a new user account associated with the specified email address and
- * password.
+ * password and returns any additional user info data or credentials.
+ *
```
Comment (bojeil-google) [inline packages/firebase/externs/firebase-auth-externs.js]: Add another paragraph.

This method will be renamed to `createUserWithEmailAndPassword` replacing the existing method with the same name in the next major version change.

Do the same for the other 3.

File: packages/firebase/externs/firebase-auth-externs.js [hunk 6]
```diff
@@ -1045,6 +1045,9 @@ firebase.auth.Auth.prototype.useDeviceLanguage = function() {};
  * Creates a new user account associated with the specified email address and
  * password and returns any additional user info data or credentials.
  *
+ * This method will be renamed to createUserWithEmailAndPassword replacing
```
Comment (bojeil-google) [inline packages/firebase/externs/firebase-auth-externs.js]: encapsulate createUserWithEmailAndPassword with ticks ` here and below.

File: packages/firebase/externs/firebase-auth-externs.js [hunk 7]
```diff
@@ -1094,7 +1097,10 @@ firebase.auth.Auth.prototype.createUserAndRetrieveDataWithEmailAndPassword = fun
 
 /**
  * Creates a new user account associated with the specified email address and
- * password. Will be depercated, replaced by
+ * password.
+ *
+ * This method will be deprecated and will be updated to resolve with a
+ * firebase.auth.UserCredential as is returned in
```
Comment (bojeil-google) [inline packages/firebase/externs/firebase-auth-externs.js]: encapsulate firebase.auth.UserCredential with ticks `

**Pull request merged**